src/compiler/checker.ts

```diff
@@ -9195,10 +9211,13 @@ namespace ts {
             if (symbol === unknownSymbol) {
                 return errorType;
             }
+            symbol = getExpandoSymbol(symbol) || symbol;

```

Comment:
getExpandoSymbol takes care of the JS special cases that used to be in getTypeReferenceTypeWorker; then you just treat them like normal symbols afterward.